File: backend/app/schemas/suggestion.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class SuggestionCreate(BaseModel):
    module_id: uuid.UUID | None = None
    action_id: str
    related_note_id: uuid.UUID | None = None
    type: str  # new_opportunity, follow_up, connection, alert, insight, enrichment
    title: str
    summary: str
    data: dict = {}
    confidence: float | None = None
    proposed_action: str  # create_note, update_note, link_notes, notify
    proposed_payload: dict = {}


class SuggestionResponse(BaseModel):
    id: uuid.UUID
    module_id: uuid.UUID | None
    action_id: str
    related_note_id: uuid.UUID | None
    type: str
    title: str
    summary: str
    data: dict
    confidence: float | None
    proposed_action: str
    proposed_payload: dict
    status: str
    snoozed_until: datetime | None
    resolved_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class SuggestionEdit(BaseModel):
    proposed_payload: dict

File: frontend/src/types/suggestion.ts
export interface Suggestion {
  id: string;
  module_id: string | null;
  action_id: string;
  related_note_id: string | null;
  type: "new_opportunity" | "follow_up" | "connection" | "alert" | "insight" | "enrichment";
  title: string;
  summary: string;
  data: Record<string, unknown>;
  confidence: number | null;
  proposed_action: string;
  proposed_payload: Record<string, unknown>;
  status: string;
  snoozed_until: string | null;
  resolved_at: string | null;
  created_at: string;
}
